src/main/java/it/gov/pagopa/swclient/mil/termsandconds/resource/TermAndCondsResource.java

```diff
@@ -0,0 +1,228 @@
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.resteasy.reactive.ClientWebApplicationException;
+
+import io.quarkus.logging.Log;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.tuples.Tuple2;
+import it.gov.pagopa.swclient.mil.bean.CommonHeader;
+import it.gov.pagopa.swclient.mil.bean.Errors;
+import it.gov.pagopa.swclient.mil.termsandconds.ErrorCode;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.SaveNewCardsResponse;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.SessionRequest;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.SessionResponse;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.TCHeaderParams;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.TCVersion;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.TcPathParam;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.TokenBody;
+import it.gov.pagopa.swclient.mil.termsandconds.bean.TokenResponse;
+import it.gov.pagopa.swclient.mil.termsandconds.dao.TCEntity;
+import it.gov.pagopa.swclient.mil.termsandconds.dao.TCRepository;
+
+
+@Path("/acceptedTermsConds")
+public class TermAndCondsResource {
+	
+	/*
+	 * Used to call the Session service to retrieve the 
+	 * taxCode and the termsAndCondsAccepted values
+	 */
+	@RestClient
+	private SessionService sessionService;
+	
+	@RestClient
+	private TokensService tokenService;
+	
+	@RestClient
+	private PmWalletService pmWalletService;
+	
+	@Inject
+	private TCRepository tcRepository;
+	
+	@ConfigProperty(name="terms.conds")
+	private String tcVersion;
+
+	@GET
+	@Path("/{taxCode}")
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Uni<Response> getTermsAndConds(@Valid @BeanParam CommonHeader headers,@Valid TcPathParam pathParams) {
+		Log.debugf("getTermsAndConds - Input parameters: %s, taxCode: %s", headers, pathParams.getTaxCode());
+		
+		return manageTokenResponse(pathParams.getTaxCode())
+					.chain(f -> manageFindVersionByTaxCode(f.getItem1().getToken()))
+					.onFailure().transform(t-> 
+					{
+						if (t instanceof NotFoundException) {
+						Log.errorf(t, "Version not found in the DB for taxCode %s",pathParams.getTaxCode());
+						return new NotFoundException(Response
+								.status(Status.NOT_FOUND)
+								.entity(new Errors(List.of(ErrorCode.ERROR_VERSION_NOT_FOUND_SERVICE)))
+								.build());
+						} else {
+							Log.errorf(t, "[%s] Internal server errorError calling tokenizator service ", ErrorCode.ERROR_VERSION_NOT_FOUND_SERVICE);
```

Comment:
The error code should be "ERROR_VERSION_SERVICE" and not "ERROR_VERSION_NOT_FOUND_SERVICE".